fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/ShowAction.java

```diff
@@ -212,18 +214,25 @@ public Object show_data(HttpServletRequest request, HttpServletResponse response
 
     @RequestMapping(path = "/api/show_table_data", method = RequestMethod.GET)
     public Object show_table_data(HttpServletRequest request, HttpServletResponse response) {
-        if (Config.enable_all_http_auth) {
-            executeCheckPassword(request, response);
-            checkGlobalAuth(ConnectContext.get().getCurrentUserIdentity(), PrivPredicate.ADMIN);
-        }
+        executeCheckPassword(request, response);
 
         String dbName = request.getParameter(DB_KEY);
         String tableName = request.getParameter(TABLE_KEY);
+
+        if (StringUtils.isEmpty(dbName) && StringUtils.isEmpty(tableName)) {
+            return ResponseEntityBuilder.okWithCommonError("dbName and tableName cannot be empty at the same time");
+        }
+
         String singleReplica = request.getParameter(SINGLE_REPLICA_KEY);
         boolean singleReplicaBool = Boolean.parseBoolean(singleReplica);
         Map<String, Map<String, Long>> oneEntry = Maps.newHashMap();
+        UserIdentity user = ConnectContext.get().getCurrentUserIdentity();
```

Comment:
Add `if (Config.enable_all_http_auth)`
Only check auth when enable_all_http_auth is true